### ISSUE TITLE
Codex message.proto doesn't match IPFS BITSWAP.md

### ISSUE THREAD

**Issue by Bulat-Ziganshin** [status-im/nim-codex#241]
We incorrectly assigned code 2 to the payload field instead of code 3:

Our code: https://github.com/status-im/nim-codex/blob/8c5939252607f74b60a49fee4570734ffa06b11c/codex/blockexchange/protobuf/message.proto#L50-L52

While [original](https://github.com/ipfs/specs/blob/main/BITSWAP.md#bitswap-120-wire-format) code: https://github.com/ipfs/go-bitswap/blob/0fa397581ca6197c6c4ca17842719370f6596e95/message/pb/message.proto#L43-L45

**Comment by dryajov** [comment]
Could you please do a PR to fix it, seems like a straightforward change that shouldn't touch anything else...

**Comment by Bulat-Ziganshin** [comment]
of course, just made check that the change wasn't intentional

**Comment by dryajov** [comment]
> of course, just made check that the change wasn't intentional

Nope, it's just a typo - tho perhaps the change actually happened in the original bitswap? Nevertheless, no reason not to make it match the original definition.

**Comment by Bulat-Ziganshin** [comment]
BitSwap 1.0 used value 2 for the field that is no more used in BitSwap 1.1:

https://github.com/ipfs/go-bitswap/blob/0fa397581ca6197c6c4ca17842719370f6596e95/message/pb/message.proto#L42-L43

And when we copied this message format, we just assigned sequential values to fields that remained in BitSwap 1.1. I.e. it's a pretty usual mistake for inattention. I caught it only because I was curious whether we use only a similar message format or target 100% message compatibility.

**Comment by dryajov** [comment]
> Member

weird, I know it was a straight up copy paste, but I guess even then it's possible to screw things up :)

**Comment by Bulat-Ziganshin** [comment]
But going deeper - if we target BitSwap compatibility, I recommend using higher values for our new fields. The reason is that if BitSwap will add new fields, they probably will reuse the same values that we are already using. F.e. we can assign our field numbers starting from 100.

**Comment by dryajov** [comment]
> But going deeper - if we target BitSwap compatibility, I recommend using higher values for our new fields. The reason is that if BitSwap will add new fields, they probably will reuse the same values that we are already using. F.e. we can assign our field numbers starting from 100.

We aren't trying to be byte level compatible, in fact we'll probably deviate from it some more once we start adding bandwidth incentives. But we can definitely use higher field numbers just to make the transition easier. So, I agree.

**Comment by Bulat-Ziganshin** [comment]
I checked protobuf docs - field numbers >=16 will use 1 more byte for encoding. Is it OK for us?

We can also start with 15 and go down.
